tools/buildmgr/cbuildgen/config/IAR.8.50.6.cmake

```diff
@@ -0,0 +1,157 @@
+# Version: 0.10.4
+# Date: 2022-01-21
+# This file maps the CMSIS project options to toolchain settings.
+#
+#   - Applies to toolchain: IAR ARM C/C++ Compiler V8.50.6
+
+############### EDIT BELOW ###############
+# Set base directory of toolchain
+set(TOOLCHAIN_ROOT)
+set(EXT)
+
+############ DO NOT EDIT BELOW ###########
+
+set(AS ${TOOLCHAIN_ROOT}/iasmarm${EXT})
+set(CC ${TOOLCHAIN_ROOT}/iccarm${EXT})
+set(CXX ${TOOLCHAIN_ROOT}/iccarm${EXT})
+set(LD ${TOOLCHAIN_ROOT}/ilinkarm${EXT})
+set(AR ${TOOLCHAIN_ROOT}/iarchive${EXT})
+set(OC ${TOOLCHAIN_ROOT}/ielftool${EXT})
+
+# Core Options
+
+if(CPU STREQUAL "Cortex-M0")
+  set(IAR_CPU "Cortex-M0")
+elseif(CPU STREQUAL "Cortex-M0+")
+  set(IAR_CPU "Cortex-M0+")
+elseif(CPU STREQUAL "Cortex-M1")
+  set(IAR_CPU "Cortex-M1")
+elseif(CPU STREQUAL "Cortex-M3")
+  set(IAR_CPU "Cortex-M3")
+elseif(CPU STREQUAL "Cortex-M4")
+  set(IAR_CPU "Cortex-M4")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "Cortex-M4.fp.sp")
+  endif()
+elseif(CPU STREQUAL "Cortex-M7")
+  set(IAR_CPU "Cortex-M7")
+  if(FPU STREQUAL "DP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp.dp")
+  elseif(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp.sp")
+  endif()
+elseif(CPU STREQUAL "Cortex-M23")
+  set(IAR_CPU "Cortex-M23")
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+elseif(CPU STREQUAL "Cortex-M33")
+  set(IAR_CPU "Cortex-M33")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp")
+  endif()
+  if(DSP STREQUAL "NO_DSP")
+    set(IAR_CPU "${IAR_CPU}.no_dsp")
+  endif()
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+elseif(CPU STREQUAL "Cortex-M35P")
+  set(IAR_CPU "Cortex-M35P")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp")
+  endif()
+  if(DSP STREQUAL "NO_DSP")
+    set(IAR_CPU "${IAR_CPU}.no_dsp")
+  endif()
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+endif()
+if(NOT DEFINED IAR_CPU)
+  message(FATAL_ERROR "Error: CPU is not supported!")
+endif()
+
+# Assembler
+
+set(ASM_CPU "--cpu ${IAR_CPU}")
+set(ASM_FLAGS)
+
+foreach(ENTRY ${DEFINES})
+  string(APPEND ASM_DEFINES "-D${ENTRY}=1 ")
+endforeach()
+
+if(BYTE_ORDER STREQUAL "Little-endian")
+  set(IAR_BYTE_ORDER "little")
+elseif(BYTE_ORDER STREQUAL "Big-endian")
+  set(IAR_BYTE_ORDER "big")
+endif()
+
+set(ASM_BYTE_ORDER "--endian ${IAR_BYTE_ORDER}")
+
+# C Compiler
+
+set(CC_CPU "--cpu=${IAR_CPU}")
+set(CC_FLAGS "--silent")
+set(CC_BYTE_ORDER "--endian=${IAR_BYTE_ORDER}")
+set(_PI "-include ")
+
+foreach(ENTRY ${DEFINES})
+  string(APPEND CC_DEFINES "-D${ENTRY} ")
+endforeach()
+
+if(SECURE STREQUAL "Secure")
+  set(CC_SECURE "--cmse")
+else()
+  set(CC_SECURE "")
+endif()
+
+# C++ Compiler
+
+set(CXX_CPU "${CC_CPU}")
+set(CXX_DEFINES "${CC_DEFINES}")
+set(CXX_BYTE_ORDER "${CC_BYTE_ORDER}")
+set(CXX_SECURE "${CC_SECURE}")
```

Comment:
Does the IAR compiler have an equivalent to `--cmse-implib` `--out-implib` (GCC) or `--import-cmse-lib-out` (AC6) to generate an import library/object with secure image entry points? If yes please add a section for generating `${OUT_DIR}/${TARGET}_CMSE_Lib.o`
For example:
https://github.com/Open-CMSIS-Pack/devtools/blob/main/tools/buildmgr/cbuildgen/config/GCC.10.2.1.cmake#L172-L174